solutions/src/test/java/ru/sberbank/school/task13/cacheproxy/CacheProxyTest.java

```diff
@@ -0,0 +1,78 @@
+import org.junit.jupiter.api.Test;
+import ru.sberbank.school.task13.cacheproxy.cache.CacheProxy;
+import ru.sberbank.school.task13.cacheproxy.exception.CacheProxyException;
+import ru.sberbank.school.task13.cacheproxy.model.CacheTest;
+import ru.sberbank.school.task13.cacheproxy.model.CacheTestImpl;
+import ru.sberbank.school.task13.cacheproxy.utils.CacheProxyUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+public class CacheProxyTest {
+    private static String dir;
+    private static final CacheProxy cacheProxyImpl = new CacheProxy("");
+
+    @BeforeAll
+    public static void init() throws IOException {
+        dir = String.valueOf(Files.createTempDirectory(""));
+    }
+
+    @Test
+    public void testMemoryCache() {
+        CacheTest proxyInstance = (CacheTest) cacheProxyImpl.cache(new CacheTestImpl("MUR"));
+        String resultFirstCall = proxyInstance.getResultUseMemoryCache();
+        String resultSecondCall = proxyInstance.getResultUseMemoryCache();
+        Assertions.assertEquals(resultFirstCall, resultSecondCall);
+        Assertions.assertEquals(1, proxyInstance.getCallCounter());
+    }
+
+    @Test
+    public void testFileCache() throws CacheProxyException {
+        CacheTest proxyInstance = (CacheTest) cacheProxyImpl.cache(new CacheTestImpl("MUR"));
+        String resultFirstCall = proxyInstance.getResultUseFileCache();
+        String resultSecondCall = proxyInstance.getResultUseFileCache();
+        Assertions.assertEquals(resultFirstCall, resultSecondCall);
+        Assertions.assertEquals(1, proxyInstance.getCallCounter());
+        String key = CacheProxyUtils.getKey(new Class[] {}, "cache_type", "getResultUseFileCache", null);
+        File file = new File(dir + File.separator + key);
+        Assertions.assertTrue(file.exists());
+        file.deleteOnExit();
+    }
+
+    @Test
+    public void testZipFile() throws CacheProxyException {
+        CacheTest proxyInstance = (CacheTest) cacheProxyImpl.cache(new CacheTestImpl("MUR"));
+        proxyInstance.zipFileTest();
+        String key = CacheProxyUtils.getKey(new Class[] {}, "", "zipFileTest", null);
+        File file = new File(dir + File.separator + key + ".zip");
+        Assertions.assertTrue(file.exists());
+        file.deleteOnExit();
+    }
+
+    @Test
+    public void testCutList() {
+        CacheTest proxyInstance = (CacheTest) cacheProxyImpl.cache(new CacheTestImpl("MUR"));
+        List result = proxyInstance.createList(500);
+        Assertions.assertEquals(100, result.size());
+    }
+
+    @Test
+    public void testKeyGenerationWhenSpecifiedArguments() {
+        CacheTest proxyInstance = (CacheTest) cacheProxyImpl.cache(new CacheTestImpl("MUR"));
+        proxyInstance.stringSize("STR", 100);
+        proxyInstance.stringSize("STR", 999);
+        Assertions.assertEquals(1, proxyInstance.getCallCounter());
+    }
+
+    @AfterAll
+    public static void finish(){
```

Comment:
WhitespaceAround: &#39;{&#39; is not preceded with whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-587431793>*